woocommerce/woocommerce
# Add product categories datastore

Fixes #33464.
Fixes #33465.
Fixes #33466.

### All Submissions:






### Changes proposed in this Pull Request:

Creates a data store with selectors and dispatchers for manipulating Product Categories through the WooCommerce API.

### How to test the changes in this Pull Request:

All these steps are console commands to be run from the WooCommerce home page after completing initial setup.

1. Create some product categories, e.g. `wp.data.dispatch('experimental/wc/admin/products/categories').createProductCategory({ name: 'Test 1', slug: 'test1' })`
2. Test the bulk fetch selector (you will need to run this twice): `console.table(wp.data.select('experimental/wc/admin/products/categories').getProductCategories({}))`
3. Test the single selector against one of the ids you saw in the bulk fetch results: `wp.data.select('experimental/wc/admin/products/categories').getProductCategory(18)`
4. Test the update action, e.g.: `wp.data.dispatch('experimental/wc/admin/products/categories').updateProductCategory(18,{ menu_order: 2 })`
5. Rerun bulk and single fetches to verify the update happened.
6. Test the delete action, e.g.: `wp.data.dispatch('experimental/wc/admin/products/categories').deleteProductCategory(18)`
7. Rerun bulk and single fetches to verify the delete happened.

#### NOTE:

The product category API spec contains an `image` property which has its own [schema and rules](https://woocommerce.github.io/woocommerce-rest-api-docs/#product-categories), but beyond defining them in data/src/product-categories/types.ts no attempt has been made to enforce them, nor to support creating a category with an image or updating a product category image.

### Other information:




### FOR PR REVIEWER ONLY:
